app/controllers/orders_controller.rb

```diff
@@ -25,18 +28,93 @@ def empty_cart
   end
 
   def checkout
+    if @cart.check_branch_inventory?
+      @cart.update(order_params)
```

Comment:
create a service. OrderCheckout and move all this payment related logic inside that service.
From controller you should only call the service(1~2 lines max)

Also these all updates should be inside transaction

Reply:
Done